src/Bootstrappers/FilesystemTenancyBootstrapper.php

```diff
@@ -54,23 +54,22 @@ public function bootstrap(Tenant $tenant)
         }
 
         // Storage facade
+        Storage::forgetDisk($this->app['config']['tenancy.filesystem.disks']);
+
         foreach ($this->app['config']['tenancy.filesystem.disks'] as $disk) {
-            /** @var FilesystemAdapter $filesystemDisk */
-            $filesystemDisk = Storage::disk($disk);
-            $this->originalPaths['disks'][$disk] = $filesystemDisk->getAdapter()->getPathPrefix();
+            $root = $this->app['config']->get("filesystems.disks.{$disk}.root");
+            $this->originalPaths['disks'][$disk] = $root;
 
-            if ($root = str_replace(
+            if (!($root = str_replace(
```

Comment:
@codeskills could you explain the changes a bit?

Reply:
I'd like to make sure this uses the root overrides correctly, as described here https://tenancyforlaravel.com/docs/v2/filesystem-tenancy/

I think I might add some assertions to the `BootstrapperTest`, assuming the PR can be set up locally

Reply:
I think this was an IDE suggestion to negate the condition and remove the original `if` part. Because the original `if` part of the conditional (the `setPathPrefix` call) was no longer needed.

Reply:
It is just a simplification, it works the same way as before, since there is `$this->app['config']["filesystems.disks.{$disk}.root"] = $root;` at the end of the function. You dont have to do this `$filesystemDisk->getAdapter()->setPathPrefix($finalPrefix = $root);` because it changes the config itself.

Reply:
Pushed that those test changes now

Reply:
I'd rewrite the if to not include the negation and assignment in the condition. You can make them separate

Reply:
You can rewrite it to something like this, to make it more clear:
```php
$root = str_replace(
    '%storage_path%',
    storage_path(),
    $this->app['config']["tenancy.filesystem.root_override.{$disk}"] ?? ''
);
// Default root
if (!$root) {
    $root = "{$root}/{$suffix}";
}
```